From c873991975c4bf60004882d16db846bb3ada6478 Mon Sep 17 00:00:00 2001
From: Weizhao Ouyang <o451686892@gmail.com>
Date: Fri, 9 Feb 2024 14:15:31 +0800
Subject: [PATCH] Fix Darwin uuid_t typo

Signed-off-by: Weizhao Ouyang <o451686892@gmail.com>
---
 scripts/mod/file2alias.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/scripts/mod/file2alias.c b/scripts/mod/file2alias.c
index 4829680a0a6d..9289166515e9 100644
--- a/scripts/mod/file2alias.c
+++ b/scripts/mod/file2alias.c
@@ -42,7 +42,11 @@ typedef struct {
 
 typedef struct {
 	__u8 b[16];
-} uuid_t;
+} compat_uuid_t;
+
+#ifdef __APPLE__
+#define uuid_t compat_uuid_t
+#endif
 
 #define	UUID_STRING_LEN		36
 
-- 
2.39.3 (Apple Git-145)

